CMakeLists.txt

```diff
@@ -238,6 +238,7 @@ endif()
 if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "AMD64")
 	if(APPLE OR UNIX)
 		set(SIMD_FLAGS_LIST "-mfma;-mavx2")
+		enable_cxx_compiler_flag_if_supported("-mpopcnt")
```

Comment:
Will this be problematic for users on pre-2007 hardware (where popcount x85_64 instructions don't exist) in the precompiled binaries? Or do you think dropping support for things that old is an OK tradeoff?

Reply:
I think you're right and anything without popcnt instruction will crash. Maybe we can compile both flavours (with and without intrinsic popcnt) and select on startup or first call to functions which one to use. For example we can use existing `is_avx2_supported()` function and in case avx2 is supported use the intrinsic popcnt. I guess we'd be leaving some CPUs (where popcnt is supported) out but we would be on the safe side, right?